### PR TITLE
MINOR: update upgrade notes with regard to KIP-149

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @bbejeck

**Comment by bbejeck** [comment]
Not sure if we should add more details, but otherwise LGTM.

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -34,7 +34,9 @@ <h1>Upgrade Guide and API Changes</h1>
     </div>
 
     <p>
-        If you want to upgrade from 1.0.x to 1.1.0 you don't need to make any code changes as the public API is fully backward compatible.
+        If you want to upgrade from 1.0.x to 1.1.0 you don't need to make any code changes as the public API is fully backward compatible in Java 7.
+        If you are using Java8 lambdas in your Kafka Streams code you might need to update your code to resolve method ambiguties.
```
Comment (guozhangwang) [inline docs/streams/upgrade-guide.html]: Java8 lambdas -> should this be `method references`?

**Comment by mjsax** [comment]
Updated this.

**Comment by mjsax** [comment]
@bbejeck What do you have in mind? It's just a hint that hot-swap won't work -- people will see if it does not compile and should be able to figure out IMHO. But happy to add more details. Suggestions?

**Comment by bbejeck** [comment]
I was thinking about that fact that is around `method references`, but the words escaped me ATM The comment from @guozhangwang covers it.

**Comment by mjsax** [comment]
Retest this please.

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -34,7 +34,9 @@ <h1>Upgrade Guide and API Changes</h1>
     </div>
 
     <p>
-        If you want to upgrade from 1.0.x to 1.1.0 you don't need to make any code changes as the public API is fully backward compatible.
+        If you want to upgrade from 1.0.x to 1.1.0 you don't need to make any code changes as the public API is fully backward compatible in Java 7.
```
Comment (dguy) [inline docs/streams/upgrade-guide.html]: I think we should move the Java 7 bit to the start of the sentence. i.e, `If you are using Java 7 then ... `

**Comment by mjsax** [comment]
Updated

**Comment by mjsax** [comment]
Retest this please.

**Review by dguy — APPROVED** [review]
Thanks @mjsax, LGTM

**Pull request merged**